website/pizzas/models.py

```diff
@@ -126,6 +127,8 @@ def save(self, *args, **kwargs):
     def delete(self, using=None, keep_parents=False):
         if self.end_reminder is not None and not self.end_reminder.sent:
             self.end_reminder.delete()
+        if self.payment is not None and not self.payment.batch:
```

Comment:
```suggestion
        if self.payment is not None and not (self.payment.batch and self.payment.batch.processed):
```

Reply:
Nice catch, I should not have committed this code